clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -2080,6 +2080,36 @@ void AtomicFunctionRule::runRule(const MatchFinder::MatchResult &Result) {
 
 REGISTER_RULE(AtomicFunctionRule, PassKind::PK_Migration)
 
+void ZeroLengthArrayDetectorRule::registerMatcher(MatchFinder &MF) {
+  MF.addMatcher(typeLoc(loc(constantArrayType())).bind("ConstantArrayType"),
+                this);
+}
+void ZeroLengthArrayDetectorRule::runRule(
+    const MatchFinder::MatchResult &Result) {
```

Comment:
Please add comment of the rule: 1. emit warnning when zero length array in device code and it is not share local memroy.